src/shapes/jolt_shape_impl_3d.cpp

```diff
@@ -190,7 +190,65 @@ JPH::ShapeRefC JoltShapeImpl3D::without_custom_shapes(const JPH::Shape* p_shape)
 		case JoltCustomShapeSubType::OVERRIDE_USER_DATA:
 		case JoltCustomShapeSubType::DOUBLE_SIDED: {
 			// Replace unsupported decorator shapes with the inner shape
-			return static_cast<const JPH::DecoratedShape*>(p_shape)->GetInnerShape();
+
+			const auto* shape = static_cast<const JPH::DecoratedShape*>(p_shape);
+			const auto* inner_shape = shape->GetInnerShape();
+
+			return without_custom_shapes(inner_shape);
+		}
+
+		case JPH::EShapeSubType::StaticCompound: {
+			const auto* shape = static_cast<const JPH::StaticCompoundShape*>(p_shape);
+
+			JPH::StaticCompoundShapeSettings settings;
+
+			for (const auto& sub_shape : shape->GetSubShapes()) {
+				settings.AddShape(
+					sub_shape.GetPositionCOM(),
```

Comment:
```sub_shape.GetPositionCOM()```

needs to be

```shape->GetCenterOfMass() + sub_shape.GetPositionCOM() - sub_shape.GetRotation() * sub_shape.mShape->GetCenterOfMass()```

because SubShapeSettings takes the model space position and internally converts to COM space.

Reply:
Ah, thank you! The moment I typed that I thought to myself that I should look into what the `COM` part was all about, but I was in too much of a hurry and forgot.